execute/plugin.go

```diff
@@ -215,15 +217,16 @@ func (p *Plugin) Observation(
 
 				// Read messages for each range.
 				for _, seqRange := range ranges {
-					msgs, err := p.ccipReader.MsgsBetweenSeqNums(ctx, selector, seqRange)
+					// TODO: check if srcChain is supported.
+					msgs, err := p.ccipReader.MsgsBetweenSeqNums(ctx, srcChain, seqRange)
```

Comment:
It seems like MsgsBetweenSeqNums will return an error if the chain is not supported. I don't think we'd want to error out from the whole observation in that case.

Maybe we need to define a specific error type in the ccip reader, e.g `ErrChainNotSupported` so that we can do `errors.Is(err, reader.ErrChainNotSupported)` and just continue in that case.

Reply:
Yes, errors are returned a bit too aggressively right now. This one in particular.

I like the `ErrChainNotSupported` idea.

Reply:
Created a ticket: https://smartcontract-it.atlassian.net/browse/CCIP-3071